File: pkg/codegen/testing/test/testdata/terraform-child-module-pp/nodejs/terraform-child-module.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as aws_s3_static_website_bucket from "./modules/aws-s3-static-website-bucket";

export class Index extends pulumi.ComponentResource {
    public readonly s3Bucket: aws.s3.Bucket;
    public readonly childModule: aws_s3_static_website_bucket.Index;

    constructor(name: string, args: IndexArgs, opts: pulumi.ComponentResourceOptions = {}) {
        super("pkg:index:component", name, args, opts);

        // Local Terraform Child Module
        this.s3Bucket = new aws.s3.Bucket("s3Bucket", {bucket: "s3BucketId"}, {
            parent: this,
        });
        this.childModule = new aws_s3_static_website_bucket.Index("childModule", {customModuleParameter: this.s3Bucket.id}, {
            parent: this,
        });
    }
}

export interface IndexArgs {
}

File: pkg/codegen/testing/test/testdata/terraform-child-module-pp/python/terraform-child-module.py
import pulumi
import modules.aws_s3_static_website_bucket as aws_s3_static_website_bucket
import pulumi_aws as aws

class IndexArgs:
    def __init__(
        self,
    ):
        pass

class Index(ComponentResource):
    def __init__(name: string, args: IndexArgs, opts: ResourceOptions = None):
        super().__init__("pkg:Index:component", name, {}, opts)

        # Local Terraform Child Module
        s3_bucket = aws.s3.Bucket("s3Bucket", bucket="s3BucketId")
        child_module = aws_s3_static_website_bucket.Index("childModule", custom_module_parameter=s3_bucket.id)
